tfenv install fails on a clean machine: lock mkdir ENOENT misreported as "Another process is installing"

# Title

`tfenv install` fails on a clean machine: lock `mkdir` ENOENT misreported as "Another process is installing"

# Body

## Summary

`tfenv install <version>` fails on any machine where `${TFENV_CONFIG_DIR}` does not yet exist (i.e., a fresh install of `tfenv` itself, before any version has been installed). The locking code introduced in v3.1.0 cannot distinguish `mkdir` failing with `ENOENT` (parent directory missing) from `EEXIST` (lock already held), and it suppresses stderr, so the user sees:

```
tfenv: Another process is installing Terraform v1.13.5. Waiting...
tfenv: Lock wait timeout after 60s. Removing stale lock and retrying.
[ERROR] Failed to acquire install lock for Terraform v1.13.5
```

…even though no other process exists. The first `tfenv install` on a brand-new install always reproduces this.

## Affected versions

- v3.1.0 (introduced)
- v3.2.0 (still present)

v3.0.0 had no locking and was unaffected.

## Reproduction

On a clean macOS machine via Homebrew (the Homebrew formula sets `TFENV_CONFIG_DIR=${XDG_CONFIG_HOME:-${HOME}/.config}/tfenv`):

```sh
brew install tfenv
rm -rf ~/.config/tfenv      # ensure clean state
tfenv install 1.13.5
```

Result: ~60 s of "Another process is installing… Waiting", then a forced retry, then `Failed to acquire install lock`. No other `tfenv` process is running.

The same reproduces on Linux if you set `TFENV_CONFIG_DIR` to a path whose parent does not exist.

## Root cause

`libexec/tfenv-install` (v3.2.0) acquires the lock with:

```bash
declare lockdir="${TFENV_CONFIG_DIR}/.install-lock-${version}";
...
while ! mkdir "${lockdir}" 2>/dev/null; do
  if [ "${lock_retries}" -eq 0 ]; then
    log 'info' "Another process is installing Terraform v${version}. Waiting...";
  fi;
  ...
done;
```

`mkdir "${lockdir}"` (without `-p`) fails for two distinct reasons:

1. `EEXIST` — the lock directory already exists → legitimate held-lock case.
2. `ENOENT` — the parent `${TFENV_CONFIG_DIR}` does not exist → not a lock conflict at all.

Because stderr is redirected to `/dev/null`, the loop treats both as contention. After `lock_max_retries=60` it `rmdir`s the (still-nonexistent) lock directory, retries the same `mkdir` (which fails again with the same `ENOENT`), and dies via `log 'error'`.

The actual install code does `mkdir -p "${dst_path}"` (where `${dst_path}=${TFENV_CONFIG_DIR}/versions/${version}`) much later, so on a clean machine the parent directory is guaranteed not to exist when the lock is taken.

## Suggested fix

Ensure `${TFENV_CONFIG_DIR}` exists before attempting to acquire the lock. Minimal patch in `libexec/tfenv-install`, immediately before the `lockdir` declaration:

```bash
mkdir -p "${TFENV_CONFIG_DIR}" \
  || log 'error' "Failed to create config directory ${TFENV_CONFIG_DIR}";
```

Optional hardening for clearer diagnostics (separate from the fix above): capture and log `mkdir`'s stderr inside the lock loop so any future ENOENT-style failure is visible to the user instead of being swallowed.

## Downstream impact

Anyone with an automated setup script that does `brew install tfenv && tfenv install <version>` (CI runners, Mac onboarding scripts, fresh VMs) hits this 100% of the time after Homebrew bumped the formula to 3.2.0. We're working around it locally with `mkdir -p "${TFENV_CONFIG_DIR:-${XDG_CONFIG_HOME:-${HOME}/.config}/tfenv}"` before invoking `tfenv install`, but it would be nice for the upstream tool to be self-sufficient on a clean install.

--- a/libexec/tfenv-install
+++ b/libexec/tfenv-install
@@ -98,11 +98,21 @@ declare lockdir="${TFENV_CONFIG_DIR}/.install-lock-${version}";
 declare lock_retries=0;
 declare lock_max_retries=60;
 
+# Ensure the config dir exists so the lock mkdir can succeed on a first install.
+# Without this, mkdir fails with ENOENT and the loop below misreads it as contention.
+mkdir -p "${TFENV_CONFIG_DIR}" || log 'error' "Failed to create ${TFENV_CONFIG_DIR}";
+
 cleanup_lock() {
   rmdir "${lockdir}" 2>/dev/null || true;
 };
 
 while ! mkdir "${lockdir}" 2>/dev/null; do
+  # Distinguish "lock held" (lockdir exists) from any other mkdir failure
+  # (missing parent, permissions, full disk, etc.) so we fail loud instead
+  # of looping silently for 60s on an unrecoverable error.
+  if [ ! -d "${lockdir}" ]; then
+    log 'error' "Failed to create lock directory ${lockdir} — check that ${TFENV_CONFIG_DIR} exists and is writable";
+  fi;
   if [ "${lock_retries}" -eq 0 ]; then
     log 'info' "Another process is installing Terraform v${version}. Waiting...";
   fi;

